template/nestjs/flake.nix

```diff
@@ -0,0 +1,82 @@
+{
+  description = "Example kickstart NestJS project.";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = inputs @ {flake-parts, ...}:
+    flake-parts.lib.mkFlake {inherit inputs;}
+    {
+      systems = ["x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin"];
+      perSystem = {
+        pkgs,
+        self',
+        ...
+      }: let
+        inherit (pkgs) nodejs dockerTools buildNpmPackage stdenv;
+        inherit (dockerTools) buildImage;
+        inherit (stdenv) mkDerivation;
+        pname = "example";
+        version = "0.1.0";
+      in {
+        packages = {
+          default = mkDerivation {
+            inherit pname version;
+            nativeBuildInputs = [
+              nodejs.pkgs.npm
+            ];
+            src = ./.;
+            buildPhase = ''
+              ln -s ${self'.packages.node_modules}/node_modules node_modules
+              npm run build
+            '';
+            installPhase = ''
+              mkdir -p $out/bin
+              ln -s ${self'.packages.node_modules}/node_modules $out/node_modules
+              cp -r dist $out/dist
+
+              cat > $out/bin/${pname} <<EOF
+              #! ${pkgs.stdenv.shell} -e
+              exec "${pkgs.nodejs}/bin/node" $out/dist/main.js "\$@"
+              EOF
+
+              chmod +x $out/bin/${pname}
+            '';
+          };
+          docker = buildImage {
+            name = pname;
+            tag = version;
+            config = {
+              Cmd = "${self'.packages.default}/bin/${pname}";
+              Env = [
+                "SSL_CERT_FILE=${pkgs.cacert}/etc/ssl/certs/ca-bundle.crt"
+              ];
+            };
+          };
+          node_modules =
+            buildNpmPackage.override {inherit nodejs;}
+            {
+              inherit pname version;
+              src = ./.;
+              # $ nix shell 'nixpkgs#prefetch-npm-deps' -c prefetch-npm-deps package-lock.json
+              # ...
+              # sha256-hbwRj8D3t8bmAQBWjkvIk8oa3M2snHdCbewXc3nOZ5Y=
+              npmDepsHash = "sha256-hbwRj8D3t8bmAQBWjkvIk8oa3M2snHdCbewXc3nOZ5Y=";
+              dontNpmBuild = true;
+              installPhase = ''
+                mkdir -p $out
+                cp -r node_modules $out/node_modules
+              '';
+            };
+        };
```

Comment:
What was the reason for separating this?

You can do both the job of `node_modules` and `default` in the same job, then the launch script in `default` can just be a `writeScriptBin` or even (my preference) live in `applications.default` instead.

Reply:
At first I thought that separating `node_modules` will create some space for CI performance improvements, but after giving it some thought, I understood it'll improve only the headaches :D